components/core/src/clp_s/JsonParser.cpp

```diff
@@ -520,6 +521,317 @@ bool JsonParser::parse() {
+                                          pair.second.value().get_immutable_view<std::string>()
+                )
+                                          .value();
+                m_current_parsed_message.add_value(node_id, str);
+            } break;
+            case NodeType::ClpString: {
+                std::string encoded_str;
+                if (pair.second.value().is<clp::ir::EightByteEncodedTextAst>()) {
+                    std::string decodedValue
+                            = pair.second.value()
+                                      .get_immutable_view<clp::ir::EightByteEncodedTextAst>()
+                                      .decode_and_unparse()
+                                      .value();
+                    encoded_str = clp::ffi::validate_and_escape_utf8_string(decodedValue.c_str())
+                                          .value();
+                } else {
+                    std::string decodedValue
+                            = pair.second.value()
+                                      .get_immutable_view<clp::ir::FourByteEncodedTextAst>()
+                                      .decode_and_unparse()
+                                      .value();
+                    encoded_str = clp::ffi::validate_and_escape_utf8_string(decodedValue.c_str())
+                                          .value();
+                }
+                m_current_parsed_message.add_value(node_id, encoded_str);
+            } break;
+            case NodeType::UnstructuredArray: {
+                std::string array_str;
+                if (pair.second.value().is<clp::ir::EightByteEncodedTextAst>()) {
+                    array_str = pair.second.value()
+                                        .get_immutable_view<clp::ir::EightByteEncodedTextAst>()
+                                        .decode_and_unparse()
+                                        .value();
+                } else {
+                    array_str = pair.second.value()
+                                        .get_immutable_view<clp::ir::FourByteEncodedTextAst>()
+                                        .decode_and_unparse()
+                                        .value();
+                }
+                m_current_parsed_message.add_value(node_id, array_str);
+                break;
+            }
+            default:
+                // Don't need to add value for obj or null
+                break;
+        }
+        m_current_schema.insert_ordered(node_id);
+    }
+
+    int32_t current_schema_id = m_archive_writer->add_schema(m_current_schema);
+    m_current_parsed_message.set_id(current_schema_id);
+    m_archive_writer->append_message(current_schema_id, m_current_schema, m_current_parsed_message);
+    return;
+}
+
+bool JsonParser::parse_from_IR() {
+    std::map<std::tuple<int, NodeType>, int> id_conversion_cache;
+    m_archive_writer->add_node(-1, NodeType::Unknown, "root");
```

Comment:
I tried taking these out and running on a small ir file. It generated an archive that was much larger than with these statements and took a couple times longer. Then when I tried to extract it is used over 20gb of memory and then I shut it down, because it was about to crash my computer.